src/core/qgscolorramp.h

```diff
@@ -50,6 +50,11 @@ class CORE_EXPORT QgsColorRamp
      */
     virtual QString type() const = 0;
 
+
+    /** Inverts the ordering of the color ramp.
+     */
+    virtual void invert() = 0;
```

Comment:
I'd suggest that since inverting support isn't mandatory (eg limited random doesn't support it), that it'd be better to not make this pure virtual and have the base class implementation just do nothing.